spec/requests/rack_attack_spec.rb

```diff
@@ -113,4 +124,70 @@ def unique_user_params
       expect(response).to have_http_status(:ok)
     end
   end
+
+  context "when there have been max admin login attempts from an IP address" do
+    let(:ip) { Faker::Internet.unique.public_ip_v4_address }
+
+    before do
+      10.times do
+        post admin_session_path, params: unique_admin_params.to_query, env: { "REMOTE_ADDR" => ip }
+      end
+    end
+
+    it "response to the next attempt from the same IP includes retry-after header" do
+      post admin_session_path, params: unique_admin_params.to_query, env: { "REMOTE_ADDR" => ip }
+      expect(response).to have_http_status(:too_many_requests)
+      expect(response.headers["Retry-After"].to_i).to be > 0
+      expect(response.headers["Retry-After"].to_i).to be <= 5.minutes
+    end
+
+    it "throttles the next attempt from the same IP" do
+      post admin_session_path, params: unique_admin_params.to_query, env: { "REMOTE_ADDR" => ip }
+      expect(response).to have_http_status(:too_many_requests)
+    end
+
+    it "does not throttle an attempt from a different IP" do
+      post admin_session_path, params: unique_admin_params.to_query, env: unique_ip_env
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "does not throttle the next attempt from the same IP after some time" do
+      travel 5.minutes
+      post admin_session_path, params: unique_admin_params.to_query, env: { "REMOTE_ADDR" => ip }
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  context "when there have been max admin login attempts for a username" do
+    let(:params) { unique_admin_params.to_query }
+
+    before do
+      10.times do
```

Comment:
You can use `ArchiveConfig.RATE_LIMIT_ADMIN_LOGIN_ATTEMPTS` here instead of hard-coding the number.

Same with `ArchiveConfig.RATE_LIMIT_ADMIN_LOGIN_PERIOD.seconds` instead of `5.minutes` too.